apps/src/templates/manageStudents/ManageStudentsFamilyNameCell.jsx

```diff
@@ -1,61 +1,53 @@
+import {useDispatch} from 'react-redux';
 import ReactTooltip from 'react-tooltip';
 import _ from 'lodash';
 import i18n from '@cdo/locale';
 import {editStudent} from './manageStudentsRedux';
 import {tableLayoutStyles} from '../tables/tableConstants';
 
-class ManageStudentFamilyNameCell extends Component {
-  static propTypes = {
-    id: PropTypes.number.isRequired,
-    familyName: PropTypes.string,
-    isEditing: PropTypes.bool,
-    editedValue: PropTypes.string,
-    sectionId: PropTypes.number,
-    inputDisabled: PropTypes.bool,
+function ManageStudentFamilyNameCell({
+  id,
+  familyName,
+  isEditing,
+  editedValue,
+  inputDisabled,
+}) {
+  const dispatch = useDispatch();
 
-    //Provided by redux
-    editStudent: PropTypes.func.isRequired,
-  };
-
-  onChangeName = e => {
-    // Convert the empty string back to null before saving, so it doesn't
-    // add empty family names to students.
+  const onChangeName = e => {
+    // Avoid saving empty string to database; convert back to null if necessary
     const newValue = e.target.value || null;
-    this.props.editStudent(this.props.id, {familyName: newValue});
+    dispatch(editStudent(id, {familyName: newValue}));
   };
 
-  render() {
-    const {familyName, editedValue, inputDisabled} = this.props;
-    const tooltipId = inputDisabled ? _.uniqueId() : '';
+  const tooltipId = inputDisabled ? _.uniqueId() : '';
 
-    return (
-      <div style={tableLayoutStyles.tableText}>
-        {!this.props.isEditing && <div>{familyName}</div>}
-        {this.props.isEditing && (
-          <div>
-            <span data-for={tooltipId} data-tip>
-              <input
-                style={styles.inputBox}
-                // Since familyName is optional, explicitly prevent value from
-                // being set to undefined.
-                value={editedValue || ''}
-                onChange={this.onChangeName}
-                placeholder={i18n.familyName()}
-                aria-label={i18n.familyName()}
-                disabled={inputDisabled}
-              />
-              <ReactTooltip id={tooltipId} role="tooltip" effect="solid">
-                <div>{i18n.disabledForTeacherAccountsTooltip()}</div>
-              </ReactTooltip>
-            </span>
-          </div>
-        )}
-      </div>
-    );
-  }
+  return (
+    <div style={tableLayoutStyles.tableText}>
+      {!isEditing && <div>{familyName}</div>}
+      {isEditing && (
+        <div>
+          <span data-for={tooltipId} data-tip>
+            <input
+              id="uitest-family-name"
+              style={styles.inputBox}
+              // Because familyName is optional, allow empty string
+              value={editedValue || ''}
+              onChange={onChangeName}
+              placeholder={i18n.familyName()}
+              aria-label={i18n.familyName()}
+              disabled={inputDisabled}
+            />
+            <ReactTooltip id={tooltipId} role="tooltip" effect="solid">
```

Comment:
When does this tooltip actually show up? It doesn't look like it's conditioned on anything other than `isEditing`, but I don't see it when editing a student's name. When I sign up for a section with a teacher account, the teacher of that section doesn't have the 'Edit' option available in the dropdown menu, so I don't see how we would even get into the `isEditing` state for a teacher. Do you know?

Reply:
I can see it when I'm in a secret words login section (where I have control over the names because they're not imported). Is that maybe what's happening?

Reply:
I added a teacher to a secret words section locally. I see a tooltip with this text on the "Secret words" column, but I don't know of a way to activate it for the "Family name" column. It doesn't appear when I mouse over the edit box for a student-student, and I can't activate the edit box at all for a teacher-student.

--------------------------------------------------

Tooltip showing over the disabled "show words" button (not this component!)
![image](https://github.com/code-dot-org/code-dot-org/assets/4108328/ad89a4a6-87a5-455e-8184-de4a5b872e81)

Student edit view: no tooltip at all (good!)
![image](https://github.com/code-dot-org/code-dot-org/assets/4108328/8c8977f4-5244-4aeb-82e6-c265b57534d7)

Teacher: no way to get into edit mode in the first place
![image](https://github.com/code-dot-org/code-dot-org/assets/4108328/b4a40d27-5aa6-4cb4-9b42-4fd362153f15)

Reply:
ahhh my bad I was just looking at the family name header tooltip. Yeah I have no idea... going to look at the pr where this was added for more context

Reply:
https://github.com/code-dot-org/code-dot-org/pull/53523

Reply:
Ahhh got it. It's the 'edit all' button at the top

Reply:
<img width="868" alt="Screenshot 2023-09-12 at 4 29 06 PM" src="https://github.com/code-dot-org/code-dot-org/assets/37230822/62a37620-5e70-4ed3-bdce-a50d6a6551be">

Reply:
Aha! Thanks for indulging my curiosity. That tooltip code just read a bit weird to me. It doesn't explicitly say it's only for when the attached input is disabled, so I got to wondering if it showed up for students when it shouldn't, and then I got to wondering whether it would show up for teachers in any case, and it clearly does.

Reply:
Of course! It definitely confused me too- even just now I had to figure out what the tooltipId was doing. Maybe I'll add a comment there to make it more clear for a future eng...